cartoframes/data/dataset.py

```diff
@@ -515,20 +517,24 @@ def exists(self):
 
     def is_public(self):
         """Checks to see if table or table used by query has public privacy"""
-        public_context = get_public_context(self.context)
+        public_client = get_client_with_public_creds(self.context)
         try:
-            public_context.sql_client.send('EXPLAIN {}'.format(self.get_query()), do_post=False)
+            public_client.execute_query('EXPLAIN {}'.format(self.get_query()), do_post=False)
             return True
         except CartoException:
             return False
 
+    def _get_client(self):
+        if self._con:
+            return get_client(self._con.creds, self._con.session, self._con.version)
+
     def _create_table(self, with_lnglat=None):
-        job = self._con.batch_sql_client \
-                  .create_and_wait_for_completion(
-                      '''BEGIN; {drop}; {create}; {cartodbfy}; COMMIT;'''
-                      .format(drop=self._drop_table_query(),
-                              create=self._create_table_query(with_lnglat),
-                              cartodbfy=self._cartodbfy_query()))
+        query = '''BEGIN; {drop}; {create}; {cartodbfy}; COMMIT;'''.format(
+            drop=self._drop_table_query(),
+            create=self._create_table_query(with_lnglat),
+            cartodbfy=self._cartodbfy_query())
+
+        job = self._client.execute_long_running_query(query)
```

Comment:
I forgot [we are raising an exception](https://github.com/CartoDB/carto-python/blob/master/carto/sql.py#L236) if the job fails, we should catch it instead of checking for the job status

Reply:
:+1:

Reply:
Good catch by the way